src/domain/course-catalog/application/use-cases/get-user-overall-progress.use-case.spec.ts

```
@@ -0,0 +1,494 @@
+        expect(response.coursesProgress.totalLessons).toBe(3);
+        expect(response.coursesProgress.completedLessons).toBe(1);
+        expect(response.coursesProgress.lessonsCompletionRate).toBe(33);
+        
+        // Tracks progress
+        expect(response.tracksProgress.totalTracks).toBe(2);
+        
+        // Flashcards stats
+        expect(response.flashcardsStats.totalAnswered).toBe(2);
+        expect(response.flashcardsStats.correctAnswers).toBe(2);
+        expect(response.flashcardsStats.accuracy).toBe(100);
+        
```

ESLint

[error] Delete `········`

(prettier/prettier)